### ISSUE TITLE
rclone defines pflags outside of package main 

### ISSUE THREAD

**Issue by clee** [rclone/rclone#3792]
#### What is the problem you are having with rclone?
I'm using rclone's `fs` library in another Go program and `fs/log/log.go` is polluting my program's global flagset.

#### What is your rclone version (output from `rclone version`)
This is using the latest HEAD from master, `557edec`. You can see [the offending flag definitions](https://github.com/rclone/rclone/blob/master/fs/log/log.go#L18-L21) in log.go.

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux, 64-bit

####  Which cloud storage system are you using? (eg Google Drive)
None/irrelevant

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
None/irrelevant

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
None/irrelevant

**Comment by clee** [comment]
When I run my program with `--help`, I expect to see only the flags that I've defined. Instead, I also see:

```
      --log-file string                Log everything to this file
      --log-format string              Comma separated list of log format options (default "date,time")
      --syslog                         Use Syslog for logging
      --syslog-facility string         Facility for syslog, eg KERN,USER,... (default "DAEMON")
```

**Comment by ncw** [comment]
Which rclone packages are you importing directly?

Can I see the code?

This is fixable in the same way that I've fixed the other packages by making a sub package logflags and an options struct which will need to be passed to log unit.

**Comment by clee** [comment]
Sure, take a look at https://gitlab.com/olaris/olaris-server/blob/develop/filesystem/rclone.go (main rclone usage point in the program I'm hacking on.)

**Comment by ncw** [comment]
With help from grep I've rooted out all the rougue pflags

```
$ git grep "github.com/rclone/rclone/fs/config/flags" | egrep '\.go:' | egrep -v '(^cmd/|^vendor|_test)'
fs/config/configflags/configflags.go:	"github.com/rclone/rclone/fs/config/flags"
fs/filter/filterflags/filterflags.go:	"github.com/rclone/rclone/fs/config/flags"
fs/log/logflags/logflags.go:	"github.com/rclone/rclone/fs/config/flags"
fs/rc/rcflags/rcflags.go:	"github.com/rclone/rclone/fs/config/flags"
vfs/vfsflags/vfsflags.go:	"github.com/rclone/rclone/fs/config/flags"
vfs/vfsflags/vfsflags_unix.go:	"github.com/rclone/rclone/fs/config/flags"
```

```
$ git grep "github.com/spf13/pflag" | egrep '\.go:' | egrep -v '(^cmd/|^vendor|_test)'
fs/config/configflags/configflags.go:	"github.com/spf13/pflag"
fs/config/flags/flags.go:	"github.com/spf13/pflag"
fs/filter/filterflags/filterflags.go:	"github.com/spf13/pflag"
fs/log/logflags/logflags.go:	"github.com/spf13/pflag"
fs/rc/rcflags/rcflags.go:	"github.com/spf13/pflag"
vfs/vfsflags/vfsflags.go:	"github.com/spf13/pflag"
vfs/vfsflags/vfsflags_non_unix.go:	"github.com/spf13/pflag"
vfs/vfsflags/vfsflags_unix.go:	"github.com/spf13/pflag"
```

There is a binary with that fix in here

https://beta.rclone.org/branch/v1.50.2-086-g1f8a3448-fix-3792-pflags-beta/ (uploaded in 15-30 mins)

And a branch for you to try [fix-3792-pflags](https://github.com/rclone/rclone/blob/fix-3792-pflags/README.md)

Let me know if it works for you!

**Comment by clee** [comment]
This does indeed fix the bug for me! I look forward to it making a future release :)

**Comment by ncw** [comment]
Just seen I broke the tests... I'll fix those then merge this...

**Comment by ncw** [comment]
I've merged this to master now (sorry for the delay) which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.51
